### PR TITLE
Fixed compilation for MSVC C++17

### PR DESCRIPTION
**Pull request by JoeyGrajciar**
compiles in VisualStudio 2019
there might be a lots of warnings about not using secure functions
_CRT_SECURE_NO_WARNINGS defined will suppress that warning

#4

### CODE REVIEW

**Comment by AetherShane** [comment]
Note: _CRT_SECURE_NO_WARNING is only needed for tests due to MS extensions.

We checked the tests were with both /c++17 and /latest under MSVC 19.20

File: cevelop-workspace/scope17/src/Test.cpp [hunk 1]
```diff
@@ -63,7 +73,7 @@ void DemoFstream(){
 
 using std::filesystem::path;
 void copy_file_transact(path const & from, path const & to) {
-   path t = to.native() + ".deleteme";
+   path t = to.native() + _T(".deleteme");
```
Comment (PeterSommerlad) [inline cevelop-workspace/scope17/src/Test.cpp]: would a better fix to have

```
path t  = to + path(".deleteme")
```

? I need to check, why I used to.native here?

Comment (PeterSommerlad) [reply cevelop-workspace/scope17/src/Test.cpp]: actually

```
   path t{to};
   t += path{".deleteme"};
````
is how it should be. Can you check that?

File: cevelop-workspace/scope17/src/Test.cpp [hunk 2]
```diff
@@ -29,7 +29,17 @@ SOFTWARE.
 #include <fstream>
 #include <cstdio>
 #include <fcntl.h>
+#ifndef _MSC_VER
 #include <unistd.h>
+#define _T(str) str
```
Comment (PeterSommerlad) [inline cevelop-workspace/scope17/src/Test.cpp]: please no macro magic!

File: cevelop-workspace/scope17/src/Test.cpp (hunk 1 above)
Comment (AetherShane) [reply cevelop-workspace/scope17/src/Test.cpp]: > actually
> 
> ```
>    path t{to};
>    t += path{".deleteme"};
> ```
> 
> is how it should be. Can you check that?

Just tested and worked. Jozef is @ Lightning Talks. He can push it once he's back.

**Comment by AetherShane** [comment]
@PeterSommerlad By the way, is test_compile_time_warning_nodiscard_scope_guards intended to fail right now? It seems to fail in general (Not just MSVC) I beieve.

**Pull request merged**